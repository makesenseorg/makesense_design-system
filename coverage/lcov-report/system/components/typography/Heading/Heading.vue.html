<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for system/components/typography/Heading/Heading.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">system/components/typography/Heading</a> Heading.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/2</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;component
    :is="tag"
    class="ds-heading"
    :class="[
      `ds-heading-${size || tag}`,
      primary &amp;&amp; `ds-heading-primary`,
      soft &amp;&amp; `ds-heading-soft`
    ]"
  &gt;
    &lt;slot /&gt;
  &lt;/component&gt;
&lt;/template&gt;
&nbsp;
&lt;script&gt;
/**
 * Headings are used as the titles of each major
 * section of a page in the interface.
 *
 * @version 1.0.0
 */
export default {
  name: 'DsHeading',
  props: {
    /**
     * The heading type used for the heading.
     * @options h1|h2|h3|h4|h5|h6
     */
    tag: {
      type: String,
      default: 'h1',
      validator: value =&gt; {
        return value.match(/(h1|h2|h3|h4|h5|h6)/)
      }
    },
    /**
     * The size used for the heading.
     * @options h1|h2|h3|h4|h5|h6
     */
    size: {
      type: String,
      default: null,
<span class="fstat-no" title="function not covered" >      validator: value =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        return value.match(/(h1|h2|h3|h4|h5|h6)/)</span>
      }
    },
    /**
     * Primary style
     */
    primary: {
      type: Boolean,
      default: false
    },
    /**
     * Muted style
     */
    soft: {
      type: Boolean,
      default: false
    }
  }
}
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" src="./style.scss"&gt;
&lt;/style&gt;
&nbsp;
&lt;docs src="./demo.md"&gt;&lt;/docs&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 22 2019 22:02:51 GMT+0100 (GMT+01:00)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
