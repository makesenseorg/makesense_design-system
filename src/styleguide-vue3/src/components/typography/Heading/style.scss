.mks-heading {
  @include reset;
  @include stack-space($font-space-l, $font-space-xxxxl);
  font-family: $font-heading;
  line-height: $line-height-s;
  letter-spacing: $letter-spacing-small;
  font-weight: $font-weight-bold;
}

.mks-heading-primary {
  color: $text-color-primary;
}

.mks-heading-soft {
  color: $color-text-lighter;
}

.mks-heading-h1 {
  font-size: $font-size-xxl;
  @media #{$media-query-large} {
    font-size: $font-size-xxxl;
  }
}

.mks-heading-h2 {
  font-size: $font-size-xxl;
}

.mks-heading-h3 {
  font-size: $font-size-xl;
}

.mks-heading-h4 {
  font-size: $font-size-l;
}

.mks-heading-h5 {
  font-size: $font-size-m;
}

.mks-heading-h6 {
  font-size: $font-size-s;
}
