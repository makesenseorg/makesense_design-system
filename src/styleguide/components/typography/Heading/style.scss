.ds-heading {
	@include reset;
	@include stack-space($font-space-l, $font-space-xxxxl);
	font-family: $font-heading;
	line-height: $line-height-s;
	letter-spacing: $letter-spacing-small;
	font-weight: $font-weight-bold;
}

.ds-heading-primary {
	color: $text-color-primary;
}

.ds-heading-soft {
	color: $color-text-lighter;
}

.ds-heading-h1 {
	font-size: $font-size-xxl;
	@media #{$media-query-large} {
		font-size: $font-size-xxxl;
	}
}

.ds-heading-h2 {
	font-size: $font-size-xxl;
}

.ds-heading-h3 {
	font-size: $font-size-xl;
}

.ds-heading-h4 {
	font-size: $font-size-l;
}

.ds-heading-h5 {
	font-size: $font-size-m;
}

.ds-heading-h6 {
	font-size: $font-size-s;
}
