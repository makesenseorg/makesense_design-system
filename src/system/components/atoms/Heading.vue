<template>
  <component
    :is="tag"
    :class="`heading heading--${color} heading--${tagClass}`"
  >
    <!-- @slot The text passed to the heading -->
    <slot />
  </component>
</template>

<script>
/**
 * Headings are used in each major section of a page in the
 * interface.
 * Heading element provides an option to change the level of the heading.
 */
export default {
  name: "MksHeading",
  status: "prototype",
  release: "0.1.0",
  props: {
    /**
     * The tag used for the title, also heading level.
     * values: h1, h2, h3, h4, h5, h6
     */
    tag: {
      type: [Boolean, String],
      required: false,
      validator: function(value) {
        return ["h1", "h2", "h3", "h4", "h5", "h6"].indexOf(value) !== -1;
      }
    },
    /**
     * The class name to apply on the title.
     * values: h1, h2, h3, h4, h5, h6
     */
    tagClass: {
      type: String,
      required: false,
      validator: function(value) {
        return ["h1", "h2", "h3", "h4", "h5", "h6"].indexOf(value) !== -1;
      }
    },
    /**
     * The color of the title.
     * values : text, primary, secondary, tertiary
     */
    color: {
      type: String,
      default: "text",
      required: false,
      validator: function(value) {
        return (
          ["primary", "secondary", "text", "tertiary"].indexOf(value) !== -1
        );
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.heading {
  @include reset;
  color: $color-text;

  &--primary {
    color: $color-primary;
  }
  &--secondary {
    color: $color-secondary;
  }
  &--tertiary {
    color: $color-tertiary;
  }
}
.h1,
h1 {
  @include title-large;
}
.h2,
h2 {
  @include title-medium;
}
.h3,
h3 {
  @include title-small;
}
h4,
.h4 {
  @include title-xsmall;
}
h5,
.h5 {
  @include text-large-black;
}
h6,
.h6 {
  @include text-body-black;
}
</style>

<docs>
  ```jsx
    <mks-heading tag="h1">The quick brown fox</mks-heading>
    <mks-heading tag="h2" color="primary">The quick brown fox</mks-heading>
    <mks-heading tag="h3" color="secondary">The quick brown fox</mks-heading>
    <mks-heading tag="h4" color="tertiary">The quick brown fox</mks-heading>
    <mks-heading tag="h5">The quick brown fox</mks-heading>
    <mks-heading tag="h6">The quick brown fox</mks-heading>

  ```
</docs>
