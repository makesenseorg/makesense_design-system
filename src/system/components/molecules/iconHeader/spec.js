import { shallowMount, createLocalVue } from "@vue/test-utils";
import Comp from "./IconHeader.vue";
import Heading from "../../atoms/heading/Heading.vue";

const localVue = createLocalVue();
localVue.component("MksHeading", Heading);

describe("IconHeader.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        image: "https://via.placeholder.com/500",
        alt: "Alt text",
        title: "Title",
        titleTag: "h1",
        description: "Description",
        descriptionType: "light",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot without description", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        image: "https://via.placeholder.com/500",
        alt: "Alt text",
        title: "Title",
        titleTag: "h1",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
