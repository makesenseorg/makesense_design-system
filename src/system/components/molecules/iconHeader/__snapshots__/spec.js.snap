// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconHeader.vue matches snapshot 1`] = `
<header
  class="icon-header"
>
  <img
    alt="Alt text"
    class="icon-header__illustration"
    src="https://via.placeholder.com/500"
  />
   
  <mks-heading-stub
    class="icon-header__title"
    color="text"
    tag="h1"
  >
    Title
  </mks-heading-stub>
   
  <p
    class="icon-header__description icon-header__description--light"
  >
    Description
  </p>
   
</header>
`;

exports[`IconHeader.vue matches snapshot without description 1`] = `
<header
  class="icon-header"
>
  <img
    alt="Alt text"
    class="icon-header__illustration"
    src="https://via.placeholder.com/500"
  />
   
  <mks-heading-stub
    class="icon-header__title"
    color="text"
    tag="h1"
  >
    Title
  </mks-heading-stub>
   
  <!---->
   
</header>
`;
